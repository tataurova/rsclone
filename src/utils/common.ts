export const extend = (a, b) => ({ ...a, ...b });

export const helperForTranslate = (value) => {
    switch (value) {
        case 'Резерв':
            return 'Reserve';
        case 'Поиск на месте':
            return 'Local search';
        case 'Стоп. Проверка информации':
            return 'Stop. Information verification';
        case 'Архив':
            return 'Archive';
    }
}

export const mockFunction = () => {
    // do nothing
};
