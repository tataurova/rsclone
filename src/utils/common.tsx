export const extend = (a, b) => ({ ...a, ...b });
