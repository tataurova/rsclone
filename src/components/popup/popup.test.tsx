import * as React from 'react';
import { mount, configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import Popup from './popup';
import { mockFunction } from '../../utils/common';

configure({ adapter: new Adapter() });

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

const MockComponent = () => (<div>Test</div>);

describe('Popup', () => {
  it('Popup', () => {
    const wrapper = mount(
            <Popup
                title={'Title'}
                openPopup={true}
                setOpenPopup={mockFunction}>
                <MockComponent />
            </Popup>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
