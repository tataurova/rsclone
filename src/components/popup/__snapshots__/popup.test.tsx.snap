// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup Popup 1`] = `ReactWrapper {}`;
